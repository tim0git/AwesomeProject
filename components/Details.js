import React from 'react';
// import PropTypes from 'prop-types';
import {
  //   SafeAreaView,
  //   StyleSheet,
  // ScrollView,
  View,
  Text,
  //   StatusBar,
} from 'react-native';

function Details(props) {
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text>Details Screen</Text>
    </View>
  );
}

// Details.propTypes = {};

export default Details;
