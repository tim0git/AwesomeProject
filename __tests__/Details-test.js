import 'react-native';
import React from 'react';
import 'jest-enzyme';
import 'react-native-mock-render/mock';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

// import sinon from 'sinon';
// import {shallow, mount} from 'enzyme';
// import renderer from 'react-test-renderer';

import Details from '../components/Details';

describe('<Details >', () => {
  // jest.useFakeTimers();
  const details = <Details />;

  it('Component Renders', () => {
    const component = shallow(details);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('has text details', () => {
    const wrapper = shallow(details);
    expect(wrapper.find('Text').at(0).props().children).toBe('Details Screen');
  });
});
