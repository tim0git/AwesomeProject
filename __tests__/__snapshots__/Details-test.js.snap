// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details > Component Renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Details Screen
  </Text>
</View>
`;
